src/main/java/org/variantsync/diffdetective/variationtree/VariationTreeNode.java

```diff
@@ -0,0 +1,364 @@
+    private final List<VariationTreeNode> childOrder;
+
+    /**
+     * Creates a new node of a variation tree.
+     *
+     * The newly created node is not connected to any other nodes.
+     *
+     * The given label is not copied, so modifications to {@code label} reflect in the label of this
+     * node. It's recommended to remove all references to {@code label} after the node is fully
+     * parsed. This makes it possible to store the unmodifiable result of {@link getLabelLines}
+     * without making a copy.
+     *
+     * @param nodeType the type of this node
+     * @param fromLine the start line number of the code of {@code label}
+     * @param toLine the end line number (exclusive) of the code of {@code label}
+     * @param featureMapping the direct feature mapping of this node, has be nun null iff
+     * {@code nodeType.isConditionalAnnotation} is {@code true}
+     * @param label a list of lines used as label
+     * @see addChild
+     * @see addBelow
+     */
+    public VariationTreeNode(
+        NodeType nodeType,
+        int fromLine,
+        int toLine,
+        Node featureMapping,
+        List<String> label
+    ) {
+        super();
+
+        this.nodeType = nodeType;
+        this.from = fromLine;
+        this.to = toLine;
+        this.label = label;
+        this.featureMapping = featureMapping;
+
+        this.childOrder = new ArrayList<>();
+    }
+
+    /**
+     * Convenience constructor for creating the root of an {@link VariationTree}.
+     *
+     * <p>The newly created root has no children yet.
+     *
+     * <p>The root is a neutral annotation (i.e., its type if {@link NodeType#IF} and its feature
+     * mapping is {@code "true"}). The label of this node is empty and therefore the line numbers
+     * are {@link DiffLineNumber#InvalidLineNumber invalid}.
+     *
+     * @see addChild
+     */
+    public static VariationTreeNode createRoot() {
+        return new VariationTreeNode(
+                NodeType.IF,
+                DiffLineNumber.InvalidLineNumber,
+                DiffLineNumber.InvalidLineNumber,
+                FixTrueFalse.True,
+                new ArrayList<>()
+        );
+    }
+
+    /**
+     * Convenience constructor for creating an artifact node of an {@link VariationTree}.
+     *
+     * @param fromLine the start line number of the code of {@code label}
+     * @param toLine the end line number (exclusive) of the code of {@code label}
+     * @param label a list of lines used as label
+     * @see addBelow
+     */
+    public static VariationTreeNode createArtifact(int fromLine, int toLine, List<String> label) {
+        return new VariationTreeNode(NodeType.ARTIFACT, fromLine, toLine, null, label);
+    }
+
+    @Override
+    public VariationTreeNode getBackingNode() {
+        return this;
+    }
+
+    @Override
+    public NodeType getNodeType() {
+        return nodeType;
+    }
+
+    @Override
+    public List<String> getLabelLines() {
+        return label;
```

Comment:
Document that this is a modifiable reference?

Reply:
Oh. I actually didn't intend this to be a modifiable list as long as we don't actually need it to be.